source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -102,6 +102,18 @@ Http::FilterHeadersStatus CacheFilter::encodeHeaders(Http::ResponseHeaderMap& he
     return Http::FilterHeadersStatus::Continue;
   }
 
+  if (lookup_result_ == nullptr) {
+    // Filter chain iteration is paused while a lookup is outstanding, but the filter chain manager
+    // can still generate a local reply. One case where this can happen is when a downstream idle
+    // timeout fires, which may mean that the HttpCache isn't correctly setting deadlines on its
+    // asynchronous operations or is otherwise getting stuck.
```

Comment:
are there other cases? It looks like in the timeout case ENVOY_BUG fires, logs, increments a stat or crashes.

In what other case does this block fire?

Reply:
This makes sense, if by async you mean same-thread, but after the cache filter has yielded control.